xgb.go

```diff
@@ -137,7 +139,19 @@ func postNewConn(conn *Conn) (*Conn, error) {
 
 // Close gracefully closes the connection to the X server.
 func (c *Conn) Close() {
-	close(c.reqChan)
+	c.broadcastDone()
+	c.wg.Wait()
+	c.conn.Close()
+	c.conn = nil
+}
+
+func (c *Conn) broadcastDone() {
+	select {
+	case <-c.done:
```

Comment:
When `readResponses` hits an unrecoverable read error on line 435 and 463, it closes this channel by calling `broadcastDone`, thus signaling to all goroutines that they should shutdown. Since receive on a closed channel does not block, having:
```
select {
case <-c.done:
   cleanup()
   return
case ...:
   ...
}
```
in a goroutine will shut it down after `broadcastDone`.

I am doing a nonblocking select here to prevent `c.done` from being closed twice, since that will trigger a panic.